src/app/SharpRaven/Data/SentryRequestFactory.cs

```diff
@@ -1,271 +1,288 @@
+                    if (stringKey.StartsWith("ALL_") || stringKey.StartsWith("HTTP_"))
+                        continue;
+
+                    var value = collection[stringKey];
+                    string stringValue = value as string;
+
+                    if (stringValue != null)
+                    {
+                        // Most dictionary values will be strings and go through this path.
+                        dictionary.Add(stringKey, stringValue);
+                    }
+                    else
+                    {
+                        // HttpCookieCollection is an ugly, evil beast that needs to be treated with a sledgehammer.
+
+                        try
+                        {
+                            // For whatever stupid reason, HttpCookie.ToString() doesn't return its Value, so we need to dive into the .Value property like this.
+                            dictionary.Add(stringKey, value.Value);
+                        }
+                        catch (Exception exception)
+                        {
+                            dictionary.Add(stringKey, exception.ToString());
+                        }
+                    }
+                }
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine(exception);
+            }
+
+            return dictionary;
+        }
+
+
+        private static void TryGetHttpContextPropertyFromAppDomain()
+        {
+			if (HttpContext != null)
+                return;
+
+            try
+            {
+                var systemWeb = AppDomain.CurrentDomain
+                    .GetAssemblies()
+                    .FirstOrDefault(assembly => assembly.FullName.StartsWith("System.Web,"));
+
+                if (systemWeb == null)
+                    return;
+
+                var httpContextType = systemWeb.GetExportedTypes()
+                    .FirstOrDefault(type => type.Name == "HttpContext");
+
+                if (httpContextType == null)
+                    return;
+
+                var currentHttpContextProperty = httpContextType.GetProperty("Current",
+                                                                             BindingFlags.Static | BindingFlags.Public);
+
+                if (currentHttpContextProperty == null)
+                    return;
+
+                CurrentHttpContextProperty = currentHttpContextProperty;
+            }
+            catch (Exception exception)
+            {
+                Console.WriteLine("An error occurred while retrieving the HTTP contextproperty: {0}", exception);
+            }
+        }
+
+        private static void GetHttpContext()
+        {
+            // [Meilu] Since reflection is performance heavy, check if we have the httpcontext only once.
+            if (!CheckedForHttpContext)
+            {
+                TryGetHttpContextPropertyFromAppDomain();
+                CheckedForHttpContext = true;
+            }

```

Comment:
Removed additional setting code and placed in needed respective locations in code, see other comments.